docs/syntax.md

```diff
@@ -0,0 +1,363 @@
+    - <connector>                         # auto-generate a new connector for every parallel connection
+    - <cable>:     [<wire>-<wire>]        # specify a range of wires to attach in parallel
+    - [<connector>, ..., <connector>]     # specify multiple simple connectors to attach in parallel
+                                          # these may be unique, auto-generated, or a mix of both
+
+  ...    
+```
+
+- Each connection set is a list of components.
+- The minimum number of items is two.
+- The maximum number of items is unlimited.
+- Items must alternatingly belong to the `connectors` and the `cables` sections.
+- When a connection set defines multiple parallel connections, the number of specified `<pin>`s and `<wire>`s for each component in the set must match. When specifying only one designator, one is auto-generated for each connection of the set.
+
+### Single connections
+
+#### Connectors
+
+- `- <designator>: <int/str>` attaches a pin of the connector, referring to a pin number (from the connector's `pins` attribute) or a pin label (from its `pinlabels` attribute), provided the label is unique.
+
+- `- <designator>` is allowed for simple connectors, since they have only one pin to connect.
+For connectors with `autogenerate: true`, a new instance, with auto-generated designator, is created.
+
+#### Cables
+
+- `<designator>: <wire>` attaches a specific wire of a cable, using its number.
+
+### Multiple parallel connections
+
+#### Connectors
+
+- `- <designator>: [<pin>, ..., <pin>]`
+
+  Each `<pin>` may be:
+
+  - `<int/str>` to refer to a specific pin, using its number (from its `pins` attribute) or its label (from its `pinlabels` attribute, provided the label is unique for this connector)
+
+  - `<int>-<int>` auto-expands to a range, e.g. `1-4` auto-expands to `1,2,3,4`; `9-7` will auto-expand to `9,8,7`.
+
+  - Mixing types is allowed, e.g. `[<pin>, <pinlabel>, <pin>-<pin>, <pin>]`
+
+- `- [<designator>, ..., <designator>]`
+
+  Attaches multiple different single pin connectors, one per connection in the set.
+  For connectors with `autogenerate: true`, a new instance, with auto-generated designator, is created with every mention.
+  Auto-generated and non-autogenerated connectors may be mixed.
+
+- `- <designator>`
+
+  Attaches multiple instances of the same single pin connector, one per connectioin in the set.
+  For connectors with `autogenerate: true`, a new instance, with auto-generated designator, is created for every connection in the set.
+  Since only connectors with `pincount: 1` can be auto-generated, pin number 1 is implicit.
+
+#### Cables
+
+- `<designator>: [<wire>, ..., <wire>]`
+
+  Each `<wire>` may be:
+
+  - `<int>` to refer to a specific wire, using its number.
+  - `<int>-<int>` auto-expands to a range.
+
+
+## BOM items
+
+Connectors (both regular, and auto-generated), cables, and wires of a bundle are automatically added to the BOM,
+unless the `ignore_in_bom` attribute is set to `true`.
```

Comment:
`ignore_in_bom` is not yet implemented. See #50 and #115.

Reply:
If the ignore_in_bom feature is wanted for 0.2 it can be fairly easily merged in on its own as it is the first commit in the branch for #115 (716cc06ea9be0df11b61b8f1768e71625508a46d)